### ISSUE TITLE
what self demo

### ISSUE THREAD

**Issue by yingnierxiao** [bungle/lua-resty-route#2]
route "*/users" (function(**self**) end)   

what **self** functions? 
**self**:to ?
**self**:render?

**Comment by bungle** [comment]
Yes, I should add that. `self` functions are basically wrapping `ngx.exit` etc. functions so that we can run possible after filters etc. I hope that I do have some time with this in a future.

**Comment by yingnierxiao** [comment]
route:use "resty.route.middleware.reqargs" {}   is error now or route:use "reqargs" {}

**Comment by bungle** [comment]
@yingnierxiao, that's right, middleware should be rewritten. They are for old version. Haven't got time to do that, sorry.

**Comment by p0pr0ck5** [comment]
@bungle I know you're busy :) having docs on this would be awesome :thumbsup:

**Comment by bungle** [comment]
@p0pr0ck5 thanks, I try to get my act together and fix this. I really think there are nice things with this library and I have spend a lot of time and effort on it. I am open for contributions, as always... there are a quite good list of what could be done here: https://github.com/bungle/lua-resty-route#roadmap

**Comment by bungle** [comment]
@p0pr0ck5, ok here are some:
https://github.com/bungle/lua-resty-route#router-api
